examples/bsc-p2p/src/main.rs

```diff
@@ -0,0 +1,89 @@
+//! Example for how hook into the bsc p2p network
+//!
+//! Run with
+//!
+//! ```not_rust
+//! cargo run -p bsc-p2p
+//! ```
+//!
+//! This launch the regular reth node overriding the engine api payload builder with our custom.
+//!
+//! Credits to: <https://blog.merkle.io/blog/fastest-transaction-network-eth-polygon-bsc>
+use chain_cfg::{boot_nodes, head, bsc_chain_spec};
+use reth_discv4::Discv4ConfigBuilder;
+use reth_network::{
+    NetworkConfig, NetworkEvent, NetworkEvents, NetworkManager,
+};
+use reth_primitives::{ForkHash, ForkId};
+use reth_provider::test_utils::NoopProvider;
+use reth_tracing::{
+    tracing::info, tracing_subscriber::filter::LevelFilter, LayerInfo, LogFormat, RethTracer,
+    Tracer,
+};
+use secp256k1::{rand, SecretKey};
+use std::{
+    net::{Ipv4Addr, SocketAddr},
+    time::Duration,
+};
+use tokio_stream::StreamExt;
+
+pub mod chain_cfg;
+
+#[tokio::main]
+async fn main() {
+    // The ECDSA private key used to create our enode identifier.
+    let secret_key = SecretKey::new(&mut rand::thread_rng());
+
+    let _ = RethTracer::new()
+        .with_stdout(LayerInfo::new(
+            LogFormat::Terminal,
+            LevelFilter::INFO.to_string(),
+            "".to_string(),
+            Some("always".to_string()),
+        ))
+        .init();
+
+    // The local address we want to bind to
+    let local_addr = SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 30303);
+
+    // The network configuration
+    let net_cfg = NetworkConfig::builder(secret_key)
+        .chain_spec(bsc_chain_spec())
+        .set_head(head())
+        .listener_addr(local_addr)
+        .build(NoopProvider::default());
```

Comment:
there is https://github.com/paradigmxyz/reth/blob/9eb7d961d71757bad3efa6270c4d8c3b1d9c208d/crates/net/network/src/config.rs#L454-L454